### PR TITLE
Stabilise batch edit component

### PR DESCRIPTION
**Pull request by bp323**
This is an aggregated list of bugs / oddities I've seen during the alpha testing. Everything in this are should be treated as priority and all efforts should be made to provide rock solid front (and back end) stability.

* [x] In some occasions I run into a bug where the week tick boxes would become unresponsive and would not become ticked when clicked. You can actually see that happening in my admin intro video, the last click I did on a week when demoing batch edit. Ticking weeks should always work.

* [x] When the Dates & times tab is open in the batch edit component, hitting the Esc button should close it, just as if you clicked on the tab.

* [x] As per #317 when events are created via batch edit, all event metadata should be copied from the last previous existing event for the newly created events.

* [x] Once sorting is implemented, every single interaction with the batch edit component should trigger sorting (and probably list rendering) too, so that admins would see a true preview of what the result of the manipulation will be.

timetable_administration

### CODE REVIEW

**Comment by bp323** [comment]
Reassigning to @Coenego for review.

**Comment by bp323** [comment]
Assigning back to myself to improve copy behaviour.

**Comment by bp323** [comment]
Back to @Coenego for review.

**Comment by Coenego** [comment]
Everything looks fine, except one small issue with copying row data. Reassigning to @bp323

**Comment by bp323** [comment]
Back to @Coenego for review.

File: shared/gh/js/utils/gh.utils.js [hunk 1]
```diff
@@ -391,6 +393,35 @@ define(['exports', 'gh.utils.instrumentation', 'gh.utils.state', 'gh.utils.templ
...
+    //  BATCH EDIT  //
+    //////////////////
+
+    /**
+     * Create and return user objects found in the provided $hiddenFields container
+     *
+     * @param  {jQuery}    $hiddenFields    The container where the hidden fields to create the user objects with can be found in
+     * @return {User[]}                     Array of user objects
+     */
+    var getOrganiserObjects = exports.getOrganiserObjects = function($hiddenFields) {
```
Comment (Coenego) [inline shared/gh/js/utils/gh.utils.js]: The coverage drops since there is no unit test for this function.

File: shared/gh/js/utils/gh.utils.js [hunk 2]
```diff
@@ -269,13 +269,15 @@ define(['exports', 'gh.utils.instrumentation', 'gh.utils.state', 'gh.utils.templ
...
         }, 10);
 
-        // Fade out and remove the container after 5 seconds if it's not marked as sticky
-        /* istanbul ignore next */
-        var fadeTimeout = setTimeout(function() {
-            if (!sticky) {
-                $notification.addClass('gh-notification-fade');
-            }
-        }, 5000);
+        if (!sticky) {
```
Comment (Coenego) [inline shared/gh/js/utils/gh.utils.js]: We might need to add `/* istanbul ignore else */` here. (coverage drops)

**Comment by Coenego** [comment]
Back to @bp323 for some follow-up on the coverage part.

File: tests/qunit/tests/js/api.util.js [hunk 3]
```diff
@@ -915,5 +916,31 @@ require(['gh.core', 'gh.api.tests'], function(gh, testAPI) {
...
+        // Verify that the $hiddenFields can be successfully returned as user objects
+        var organisers = gh.utils.getOrganiserObjects($hiddenFields);
+        assert.equal(organisers[0].displayName, 'Bert Pareyn', 'Verify that the first user has the correct displayName');
+        assert.equal(organisers[0].id, '1', 'Verify that the first user has the correct ID');
+        assert.equal(organisers[1].displayName, 'Mathieu Decoene', 'Verify that the second user has the correct displayName');
+        assert.equal(organisers[1].id, undefined, 'Verify that the second user has no ID');
+
+    });
+
+
```
Comment (Coenego) [inline tests/qunit/tests/js/api.util.js]: This line can probably go

File: shared/gh/partials/admin-batch-edit-date.html [hunk 4]
```diff
@@ -41,4 +41,8 @@
         <% }); %>
         <button type="button" class="btn btn-default gh-btn-secondary gh-batch-edit-date-add-day"><i class="fa fa-plus-square"></i> Add another day</button>
     </div>
+
+    <div class="gh-batch-edit-date-processing" style="display: none;">
```
Comment (Coenego) [inline shared/gh/partials/admin-batch-edit-date.html]: Can you just add a line of comment above this block? `<!-- ... -->`

File: tests/qunit/tests/js/api.util.js [hunk 5]
```diff
@@ -915,5 +916,31 @@ require(['gh.core', 'gh.api.tests'], function(gh, testAPI) {
...
+        assert.throws(function() {
+            gh.utils.getOrganiserObjects();
+        }, 'Verify that an error is thrown when no $hiddenFields was provided');
+        // Verify that the $hiddenFields can be successfully returned as user objects
+        var organisers = gh.utils.getOrganiserObjects($hiddenFields);
+        assert.equal(organisers[0].displayName, 'Bert Pareyn', 'Verify that the first user has the correct displayName');
+        assert.equal(organisers[0].id, '1', 'Verify that the first user has the correct ID');
+        assert.equal(organisers[1].displayName, 'Mathieu Decoene', 'Verify that the second user has the correct displayName');
+        assert.equal(organisers[1].id, undefined, 'Verify that the second user has no ID');
+
```
Comment (Coenego) [inline tests/qunit/tests/js/api.util.js]: This line can probably go

**Comment by Coenego** [comment]
Left a couple of comments regarding code styling. Reassigning to @bp323

**Pull request merged**